### PR TITLE
Fix unix changing from real-time to normal policy

### PR DESCRIPTION
**Pull request by kmdade**
Thank you for authoring this very helpful library.

I noticed that nothing was happening when attempting to modify the policy of a real-time thread, and I believe I have identified the issue.

# Changes
* A unit test is added to demonstrate the previously failing behavior.
* The function `set_thead_priority_and_policy()` is updated to also call [pthread_setschedparam](https://man7.org/linux/man-pages/man3/pthread_setschedparam.3.html) when setting a "normal" policy so that the thread schedule policy is updated in addition to the nice value.

# Additional Notes
I develop on MacOS and Ubuntu, and have tested on both platforms. I admittedly did not familiarize myself with the overall cross-platform test plan, and will gladly follow any instructions. Suggested improvements to implementation or style are welcome, thank you for considering these changes.

## Summary by CodeRabbit

- **New Features**
	- Enhanced thread priority management to set static priority 0 for normal priority threads and adjust relative priority through niceness.

- **Tests**
	- Added a test to verify correct behavior when switching between realtime and normal thread policies, ensuring appropriate capabilities are required.
	- Updated and added various test scenarios related to thread policy management.

### CODE REVIEW

File: src/unix.rs [hunk 1]
```diff
@@ -596,8 +596,23 @@ pub fn set_thread_priority_and_policy(
...
+                        &params as *const libc::sched_param,
+                    )
+                };
+
+                match ret {
+                    0 => Ok(()),
+                    e => Err(Error::OS(e)),
+                }?;
+
+                // Normal priority threads adjust relative priority through niceness.
```
[the remark this answers is not in the document]
Comment (iddm) [reply src/unix.rs:615]: I think this is a good suggestion.

File: src/unix.rs [hunk 2]
```diff
@@ -820,6 +835,44 @@ mod tests {
...
+
+        // On linux, normal priority threads always have static priority 0. Instead the "nice" value is used.
+        #[cfg(not(target_os = "linux"))]
+        assert_eq!(thread.get_priority(), Ok(expected_priority));
+        #[cfg(target_os = "linux")]
+        {
+            let nice = unsafe { libc::getpriority(0, 0) };
+            assert_eq!(nice, TEST_PRIORITY as i32);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (iddm) [reply src/unix.rs:874]: No.

**Comment by iddm** [comment]
Hi, @kmdade! 
Thank you very much for the kind words and your contribution! I find it valid, and it is an overlook on my side. There is, however, only one stylistic comment, which also applies to the code above yours (mine). Please take a look at the comment made by CodeRabbit AI.

**Comment by kmdade** [comment]
The suggested change is applied.

**Review by iddm — APPROVED** [review]

**Comment by iddm** [comment]
Thank you! https://crates.io/crates/thread-priority/1.1.0

**Pull request merged**